RELEASE/scripts/autoscend/quests/level_08.ash

```diff
@@ -499,165 +469,266 @@ void theeXtremeSlopeChoiceHandler(int choice) {
 
 boolean L8_trapperNinjaLair()
 {
-	if (internalQuestStatus("questL08Trapper") != 2)
+	//adventure in the lair of the ninja snowmen to find and fight ninja snowman assassins
+	if(internalQuestStatus("questL08Trapper") != 2)
 	{
 		return false;
 	}
-
-	if(!have_skill($skill[Rain Man]) && (pulls_remaining() >= 3) && (internalQuestStatus("questL08Trapper") < 3))
+	if(get_property("auto_L8_ninjaSkip").to_boolean())
+	{
+		return false;	//this ascension we are skipping ninja route
+	}
+	if(L8_trapperPeak())
+	{
+		return true;	//unlock peak if ready to do so.
+	}
+	
+	if(isActuallyEd() && in_hardcore())
+	{
+		auto_log_info("Hardcore Ed prefers to do the extreme slope", "blue");
+		set_property("auto_L8_ninjaSkip", true);		//give up on ninja route for this ascension
+		return true;
+	}
```

Comment:
Ed could do the Ninja route (he has +20 combat from Talisman of Horus) but he needs to not be instagibbed by assassins so we would want to make sure we have enough cold resist+DA+HP for that.
However the mafia function `expected_damage()` doesn't work for assassins and the mafia dev's don't seem interested in fixing it (see https://kolmafia.us/showthread.php?25221-expected_damage()-can-be-very-wrong-in-some-cases&p=158332) which means we'd have to write our own calcs for them or as zarqon says on that thread we could use Batbrain to save us the effort.

I'd leave it as is for now, maybe add a comment and/or raise an issue for this. It's not super important for Ed as it'll save some trivial amount of turns (Shelter of Shed's -20 combat rate makes the extreme slope pretty quick for him generally).

Reply:
I see.
I wanted to write up a function that calculates expected damage for snowman assassin and incorporate it into the decision making in general, after making such a function I planned to move this from draft to pull request.
I did not realize the lack of such a function was also the issue with ed. I will add a note there about this for now. and when (or if) i make such a function I will include ed in its calculation. thank you